packages/stark-ui/src/modules/toast-notification/services/toast-notification.service.ts

```diff
@@ -0,0 +1,116 @@
+import { Observable, Observer } from "rxjs";
+import { map } from "rxjs/operators";
+import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+import {
+	starkToastNotificationServiceName,
+	StarkToastNotificationService,
+	StarkToastNotificationOptions,
+	StarkToastNotificationResult
+} from "../services";
+import {
+	StarkToastNotificationComponent,
+	StarkToastNotificationMessage
+} from "../components";
+
+/**
+ * Service to display different types of messages in a toast (info, warning or error).
+ */
+@Injectable()
+export class StarkToastNotificationServiceImpl implements StarkToastNotificationService {
+	public config: MatSnackBarConfig;
+	public template: TemplateRef<any>;
+	public obs: Observer<StarkToastNotificationResult>;
+	public toast: MatSnackBarRef<StarkToastNotificationComponent>;
+	public _defaultToastOptions: StarkToastNotificationOptions = {
+		delay: 3000,
+		position: "bottom right",
+		actionClasses: []
+	};
+
+	/**
+	 * Class constructor
+	 * @param snackBar - Tha snackBar used to display the toast
+	 * @param logger - The logger of the application
+	 * @param ref - The angular application reference running on the page
+	 */
+	public constructor(
+		@Inject(MatSnackBar) public snackBar: MatSnackBar,
+		@Inject(STARK_LOGGING_SERVICE) public logger: StarkLoggingService,
+		public ref: ApplicationRef
+	) {
+		this.logger.debug(starkToastNotificationServiceName + " loaded");
+	}
+
+
+	/**
+	 * Returns an observable that will emit one of the possible StarkToastNotificationResult after the toast is closed
+	 * @param message - Message to be shown in the toast.
+	 * @returns Observable that will emit the result value as soon as the toast is closed.
+	 */
+	public show(message: StarkToastNotificationMessage): Observable<StarkToastNotificationResult> {
+		if (this.obs && !this.obs.closed) {
+			this.obs.next(StarkToastNotificationResult.CLOSED_BY_NEW_TOAST);
+			this.obs.complete();
+		}
+		return new Observable((observer: Observer<StarkToastNotificationResult>) => {
+			this.config = this.getConfig(message);
+			this.obs = observer;
+			this.toast = this.snackBar.openFromComponent(StarkToastNotificationComponent, this.config);
+			this.toast.afterDismissed().pipe(
+				map((ret: MatSnackBarDismiss) => {
+					if (!observer.closed) {
+						if (!ret.dismissedByAction) {
+							observer.next(StarkToastNotificationResult.CLOSED_ON_DELAY_TIMEOUT);
+							this.ref.tick();
+						}
+					}
+					observer.complete();
+				})
+			).subscribe();
+		});
+	}
+	
+	/**
+	 * Hides the current toast and emits the corresponding result in the observable returned by the show() method
+	 * @param result - Result to be sent to the subscribers of the show method's observable
+	 */
+	public hide(result: StarkToastNotificationResult = StarkToastNotificationResult.ACTION_CLICKED): void {
```

Comment:
Previous API had this parameter to allow the user to send a custom result (in the example, `StarkToastNotificationResult.HIDDEN`), instead of the default `StarkToastNotificationResult.ACTION_CLICKED`.